lib/pharos/phases/configure_master.rb

```diff
@@ -322,13 +322,21 @@ def secrets_encryption_exist?
       def configure_secrets_encryption
         logger.info { "Creating secrets encryption configuration ..." }
         @ssh.exec!("sudo install -m 0700 -d #{SECRETS_CFG_DIR}")
+
         # Generate keys
-        key1 = Base64.strict_encode64(SecureRandom.random_bytes(32))
-        key2 = Base64.strict_encode64(SecureRandom.random_bytes(32))
-        cluster_context['secrets_encryption'] = {
-          'key1' => key1,
-          'key2' => key2
-        }
+        if cluster_context['secrets_encryption']
+          key1 = cluster_context['secrets_encryption']['key1']
+          key2 = cluster_context['secrets_encryption']['key2']
+        else
+          key1 = Base64.strict_encode64(SecureRandom.random_bytes(32))
```

Comment:
I think this should be split into separate phase that is executed before `ConfigureMaster`?

Reply:
It could be handled as part of `cache_kube_certs` / `copy_kube_certs`.